thirdeye-ui/src/app/components/alert-accuracy-colored/alert-accuracy-colored.component.tsx

```diff
@@ -15,54 +15,138 @@
 
 import { Typography, useTheme } from "@material-ui/core";
 import { capitalize } from "lodash";
-import React, { FunctionComponent } from "react";
+import React, { FunctionComponent, useEffect, useMemo } from "react";
 import { useTranslation } from "react-i18next";
-import { SkeletonV1 } from "../../platform/components";
+import { useInView } from "react-intersection-observer";
+import { SkeletonV1, TooltipV1 } from "../../platform/components";
+import { ActionStatus } from "../../rest/actions.interfaces";
+import { useGetAlertStats } from "../../rest/alerts/alerts.actions";
+import { AnomalyFeedbackType } from "../../rest/dto/anomaly.interfaces";
 import { getAlertAccuracyData } from "../../utils/alerts/alerts.util";
+import { LoadingErrorStateSwitch } from "../page-states/loading-error-state-switch/loading-error-state-switch.component";
 import type { AlertAccuracyColoredProps } from "./alert-accuracy-colored.interface";
 
 export const AlertAccuracyColored: FunctionComponent<AlertAccuracyColoredProps> =
-    ({
-        alertStats,
-        renderCustomLoading,
-        typographyProps,
-        renderCustomText,
-        defaultSkeletonProps,
-    }) => {
+    ({ alertId, typographyProps, label, defaultSkeletonProps, start, end }) => {
         const theme = useTheme();
         const { t } = useTranslation();
 
-        if (!alertStats) {
-            return (
-                renderCustomLoading || (
-                    <SkeletonV1 width={50} {...defaultSkeletonProps} />
-                )
-            );
-        }
+        const { alertStats, getAlertStats, status } = useGetAlertStats();
+
+        const { ref, inView } = useInView({
+            triggerOnce: true,
+            delay: 250,
+            threshold: 1,
+        });
+
+        useEffect(() => {
+            if (inView) {
```

Comment:
This component is used in other places as well (where this loading isn't an issue), so would it be wise to have this data fetching trigger in the component itself?

Reply:
I don't see an issue having this component fetch the data

Reply:
Tested it out. Should be fine.